packages/modal/src/use-modal.ts

```diff
@@ -217,5 +217,7 @@ export function useAriaHidden(
         undo?.()
       }
     }
-  }, [shouldHide, ref])
+    // Even if `ref.current` changes, `ref` does not change, so `ref.current` is specified.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should be able to restructure the code to remove the necessity to disable the eslint rule.

```tsx
const currentElement = ref.current
useEffect(() => {
  // still use ref.current in the effect, because it could have changed since triggering the effect
}, [shouldHide, currentElement])
```

Reply:
Thank you for the fast response!
I think I did not express my self clear enough.
Your first implementation was logically correct, only eslint did not like the implementation.
With my code comment I meant, that we should still use `ref.current` inside of the effect.

Full example: 
```tsx
export function useAriaHidden(
  ref: RefObject<HTMLElement>,
  shouldHide: boolean,
) {
  // save current ref in a local var to trigger the effect on identity change
  const currentElement = ref.current

  useEffect(() => {
    // keep using `ref.current` inside the effect
    // it may have changed during render and the execution of the effect
    if (!ref.current) return undefined

    let undo: Undo | null = null

    if (shouldHide && ref.current) {
      undo = hideOthers(ref.current)
    }

    return () => {
      if (shouldHide) {
        undo?.()
      }
    }
  }, [shouldHide, ref, currentElement])
}
```

Reply:
Oh, sorry, I didn't check carefully.
And I realized that I could write it more concisely. Could you please review it?

Reply:
💖